packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/activityaware/AppCompatActivityAwareHelper.kt

```diff
@@ -0,0 +1,42 @@
+package expo.modules.kotlin.activityaware
+
+import androidx.appcompat.app.AppCompatActivity
+import java.lang.ref.WeakReference
+import java.util.concurrent.CopyOnWriteArrayList
+
+/**
+ * Similar to [androidx.activity.contextaware.ContextAwareHelper]
+ *
+ * Helper class for implementing [AppCompatActivityAware].
+ */
+class AppCompatActivityAwareHelper {
+  val listeners = CopyOnWriteArrayList<OnActivityAvailableListener>()
+
+  var activity = WeakReference<AppCompatActivity>(null)
+
+  fun peekAvailableActivity(): AppCompatActivity? {
+    return activity.get()
+  }
+
+  fun addOnActivityAvailableListener(listener: OnActivityAvailableListener) {
+    val a = activity.get()
+    if (a != null) {
+      listener.onActivityAvailable(a)
+    }
```

Comment:
is it intentional to emit the event in the adding method? if yes with this side effect, please add some comments.

Reply:
Yes, it is intentional. I've adapted this class from https://android.googlesource.com/platform/frameworks/support/+/HEAD/activity/activity/src/main/java/androidx/activity/contextaware/ContextAwareHelper.java and in `AndroidX` they do it that way, but yeah, I'll add some comment 😉

Reply:
I've added a comment to the `AppCompatActivityAware` interface 😉